lib/iris/tests/__init__.py

```diff
@@ -412,13 +416,14 @@ def assertCubeDataAlmostEqual(self, cube, reference_filename, *args,
                 self.assertEqual(stats.get('masked', False),
                                        isinstance(cube.data, ma.MaskedArray))
                 nstats = np.array((stats.get('mean', 0.), stats.get('std', 0.),
-                                   stats.get('max', 0), stats.get('min', 0)))
-                if math.isnan(stats.get('mean', 0)):
-                    self.assertEqual(math.isnan(stats.get('mean', 0)),
-                                     math.isnan(cube.data.mean()))
+                                   stats.get('max', 0.), stats.get('min', 0.)),
+                                  dtype=np.float_)
+                if math.isnan(stats.get('mean', 0.)):
+                    self.assertTrue(math.isnan(cube.data.mean()))
                 else:
                     cube_stats = np.array((cube.data.mean(), cube.data.std(),
-                                           cube.data.max(), cube.data.min()))
+                                           cube.data.max(), cube.data.min()),
+                                          dtype=np.float_)
```

Comment:
A simple tidy to align what `dtype` of what is saved and what is compared i.e. `np.float_`